### PR TITLE
Automatically generate beaker sets from metadata

### PR DESCRIPTION
**Pull request by ekohl**
Modulesync 1.1 allows reading the metadata which makes it easy to read files in the directory. This allows generating the beaker sets automatically based on metadata.json.

Currently 1.1 is unreleased to this is a draft: https://github.com/voxpupuli/modulesync/pull/174

### CODE REVIEW

File: metadata2beaker.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+        options = {}
+        options[:hostname] = "#{name}.example.com" if hostname
+        # Docker messes up cgroups and modern systemd can't deal with that when
+        # PIDFile is used.
+        if pidfile
+          case os['operatingsystem']
+          when 'CentOS'
+            options[:image] = 'centos:7.6.1810' if release == '7'
+          when 'Ubuntu'
```
Comment (ekohl) [inline metadata2beaker.rb:20]: @alexjfisher I recall you also had to apply this for some Ubuntu version but I couldn't find which version. Do you recall?

**Comment by ekohl** [comment]
1.1 has been released, restarting tests.

File: metadata2beaker.rb (hunk 1 above)
Comment (alexjfisher) [reply metadata2beaker.rb:20]: https://github.com/voxpupuli/puppet-zabbix/pull/673

**Comment by mmoll** [comment]
:broken_heart:

**Comment by ekohl** [comment]
I'll pick this up after 2.1 branches. I have some ideas about this and sharing it with voxpupuli.

**Comment by ekohl** [comment]
Included in https://github.com/theforeman/foreman-installer-modulesync/pull/106

**Pull request closed without merging**